pkg/ui/src/views/queries/index.tsx

```diff
@@ -30,29 +34,57 @@ interface QueriesPageState {
   sortSetting: SortSetting;
 }
 
-const QUERIES_COLUMNS: ColumnDescriptor<CollectedStatementStatistics$Properties>[] = [
-  {
-    title: "Query",
-    className: "queries-table__col-query-text",
-    cell: (query) => query.key.query,
-    sort: (query) => query.key.query,
-  },
-  {
-    title: "Count",
-    cell: (query) => FixLong(query.stats.count).toInt(),
-    sort: (query) => FixLong(query.stats.count).toInt(),
-  },
-  {
-    title: "Avg Rows",
-    cell: (query) => Math.round(query.stats.num_rows.mean),
-    sort: (query) => query.stats.num_rows.mean,
-  },
-  {
-    title: "Avg Latency",
-    cell: (query) => Duration(query.stats.service_lat.mean * 1e9),
-    sort: (query) => query.stats.service_lat.mean,
-  },
-];
+function StatementLink(props: { statement: string }) {
+  const summary = summarize(props.statement);
+
+  return (
+    <Link to={ `/statement/${encodeURIComponent(props.statement)}` }>
+      <div title={ props.statement }>{ shortStatement(summary, props.statement) }</div>
+    </Link>
+  );
+}
+
+function shortStatement(summary: StatementSummary, original: string) {
+  switch (summary.statement) {
+    case "update": return "UPDATE " + summary.table;
+    case "insert": return "INSERT INTO " + summary.table;
+    case "select": return "SELECT FROM " + summary.table;
+    case "delete": return "DELETE FROM " + summary.table;
+    case "create": return "CREATE TABLE " + summary.table;
+    default: return original;
+  }
+}
+
+function makeQueriesColumns(queries: CollectedStatementStatistics$Properties[])
+    : ColumnDescriptor<CollectedStatementStatistics$Properties>[] {
+  const countBar = countBarChart(queries);
+  const rowsBar = rowsBarChart(queries);
+  const latencyBar = latencyBarChart(queries);
+
+  return [
+    {
+      title: "Query",
+      className: "queries-table__col-query-text",
+      cell: (query) => <StatementLink statement={ query.key.query } />,
+      sort: (query) => query.key.query,
+    },
+    {
+      title: "Count",
+      cell: countBar,
+      sort: (query) => FixLong(query.stats.count).toInt(),
+    },
+    {
+      title: "Avg Rows",
```

Comment:
maybe change this and "latency" below to "mean", since you seem to be using that everywhere else

Reply:
Done.